src/types/missing_scene.ts

```
@@ -0,0 +1,35 @@
+// export interface IMissingSceneItem {
+//   _id: string;
+//   path: string;
+// }
+
+export default class MissingScene {
+  _id: string;
+  path: string;
+  constructor(_id: string, path: string) {
+    this._id = _id;
+    this.path = path;
+  }
+}
+
+import { sceneCollection, missingSceneCollection } from "../database/index";
+import * as logger from "../logger";
+export async function purgeMissingScenes() {
+  const items = await missingSceneCollection.getAll();
+  items.forEach(async item => {
```

There really should be a try/catch encompassing all the db operations (or individually if you want custom messages) to avoid silent errors
@boi123212321 is there any particular handling that should be done if one of the operations fails ? Do we catch each promise individually to do as many as possible, or fail early ?

---

We should also log "deleting missing scene {path}", otherwise there is no way to tell what is happening

---

@leadwolf I'd let the function throw an error (so no try-catch), and handle it whereever it's called

---

But this is the root function being called: the express route `/empty-recycle-bin` does not return the result of this function. So if we don't catch here, it'll be an UnhandledPromiseRejection

---

The function call should be wrapped in try-catch then, and be awaited (it currently isn't)